libraries/anvill_passes/src/BrightenPointerOperations.cpp

```diff
@@ -524,18 +524,22 @@ PointerLifter::visitPtrToIntInst(llvm::PtrToIntInst &inst) {
   // was once again casted to be a pointer, maybe of a different type.
   // Try and propagate if possible
 
-  if (auto ptr_inst = llvm::dyn_cast<llvm::Instruction>(&inst)) {
+  if (auto ptr_inst = llvm::dyn_cast<llvm::Instruction>(inst.getOperand(0))) {
     auto [new_ptr, worked] = visitInferInst(ptr_inst, inferred_type);
     if (worked) {
       return {new_ptr, worked};
     }
   }
+  if (auto constant_ptr =
```

Comment:
Use `else if`. Also need to handle `llvm::Argument`?

Reply:
👍 Now I just check the operand for matching types if its not another instruction to visit, if not it hits the fall through and forces a cast.